src/main/java/io/github/thebusybiscuit/cscorelib2/data/PersistentUuidDataType.java

```diff
@@ -0,0 +1,50 @@
+package io.github.thebusybiscuit.cscorelib2.data;
+
+import com.google.gson.JsonObject;
+import org.bukkit.persistence.PersistentDataAdapterContext;
+import org.bukkit.persistence.PersistentDataType;
+
+import javax.annotation.Nonnull;
+import java.util.UUID;
+
+public class PersistentUuidDataType implements PersistentDataType<int[], UUID> {
+
+    public static final PersistentDataType<int[], UUID> TYPE = new PersistentUuidDataType();
```

Comment:
You should add a protected constructor, since this should be a singleton unless extended.